Userland/Libraries/LibJS/Runtime/Error.cpp

```diff
@@ -19,15 +20,22 @@ NonnullGCPtr<Error> Error::create(Realm& realm)
     return realm.heap().allocate<Error>(realm, *realm.intrinsics().error_prototype()).release_allocated_value_but_fixme_should_propagate_errors();
 }
 
-NonnullGCPtr<Error> Error::create(Realm& realm, DeprecatedString const& message)
+NonnullGCPtr<Error> Error::create(Realm& realm, String message)
 {
     auto& vm = realm.vm();
     auto error = Error::create(realm);
     u8 attr = Attribute::Writable | Attribute::Configurable;
-    error->define_direct_property(vm.names.message, PrimitiveString::create(vm, message), attr);
+    error->define_direct_property(vm.names.message, PrimitiveString::create(vm, move(message)), attr);
     return error;
 }
 
+NonnullGCPtr<Error> Error::create(Realm& realm, StringView message)
+{
+    if (auto message_as_string = String::from_utf8(message); !message_as_string.is_error())
+        return create(realm, message_as_string.release_value());
+    return InternalError::create(realm, String::from_utf8_short_string("OOM"sv));
```

Comment:
I'm not a fan of doing this here; after all `Error::create()` is a normal factory function like any other - and should propagate OOM to the caller via `ThrowCompletionOr` who can then decide what to do with that.